help_to_heat/frontdoor/interface.py

```diff
@@ -193,10 +226,38 @@ def get_session(self, session_id):
     @with_schema(load=CreateReferralSchema, dump=ReferralSchema)
     @register_event(models.Event, "Referral created")
     def create_referral(self, session_id):
-        session_data = api.session.get_session(session_id)
-        data = SupplierConverter(session_id).replace_in_session_data(session_data)
-        supplier = portal.models.Supplier.objects.get(name=data["supplier"])
-        referral = portal.models.Referral.objects.create(session_id=session_id, data=data, supplier=supplier)
+        answers = api.session.get_session(session_id)
+
+        # filter to only answers given on the user's journey
+        journey = calculate_journey(answers, confirm_and_submit_page)
```

Comment:
the idea for it being that since we don't clear answers on pressing back then certain follow ups are submitted when they shouldn't be. there are a reasonable number of corner case fixes that have needed to be programmed in in the past (loft, epc details) so it'd be nice to cover the general case for the remaining unhandled issues (income question, park home main residence question)

it's something that has been in the scope of the ticket since it was brought in & something I'm confident in. since answers are always set to the page they're submitted on, though if we don't think it's worthwhile at this time it can be dropped and raised into the backlog

Reply:
Ok, let's keep it.